### ISSUE TITLE
Run command on several terminals

### ISSUE THREAD

**Issue by exander77** [Ylianst/MeshCentral#4535]
Is there a way to run a terminal command on several terminals at once? Or use some standard utility which does that in conjunction with MeshCentral?

**Comment by dinger1986** [comment]
You could use the plugin scripttask https://github.com/ryanblenis/MeshCentral-ScriptTask or maybe look at something like Tactical RMM?

I believe @Ylianst is planning on integrating scripttask or building something in the future that will be do similar to it

**Comment by si458** [comment]
this is already built into 1.0.85 but its expermimental 
(it currently only runs and doesnt return anything)
1. select computers with checkboxes
2. select group action
3. run commands
4. select cmd or powershell
5. select if admin(system) or the logged in user
6. select a file stored in meshcentral or enter command
7. click run

**Comment by dinger1986** [comment]
awww sweet!!

**Comment by exander77** [comment]
> this is already built into 1.0.85 but its experimental (it currently only runs and doesnt return anything)
> 
>     1. select computers with checkboxes
> 
>     2. select group action
> 
>     3. run commands
> 
>     4. select cmd or powershell
> 
>     5. select if admin(system) or the logged in user
> 
>     6. select a file stored in meshcentral or enter command
> 
>     7. click run

That's exactly what I need.

**Comment by exander77** [comment]
Can We expect some improvement to get result from each node?

**Comment by dinger1986** [comment]
> Can We expect some improvement to get result from each node?

If you have a feature request it would be worth opening one and ask specifically for what you want it to do, "an improvement" is all relative :).

**Comment by si458** [comment]
> Can We expect some improvement to get result from each node?

yes @Ylianst said it was in the works but for the moment its only runs commands
he is away at the moment, but the plan he said was to merge the scripttask plugin into MC

**Comment by smartekIT** [comment]
to be honest the way it's right now is far away from perfect comparing to Script-Task plugin. Maybe that's me only who see it this way, but it's different from one to another. As a long time user of the plugin, I liked how everything is in one page with it's own clicking tabs and option under each tab but all in one page. handy to select everything from there and store the tasks there too in the same page and see outcome from each. + scripts can be organized into folders tree and ability to edit, rename, view, run and other on any script while in the same page, even can schedule it there too.

I got very excited when @Ylianst announced that he is willing to integrate the plugin as a built-in option ( Thanks to him personally and his efforts, I look at him and how he help us for free as a hero). After I've seen how it's now and imagine how it's going to look like in MC, I sort of stepped back and said to myself I keep using the plugin with that easy and well arranged interface rather than going into different pages and dialogs and menus as it's in MeshCentral.

Must admit that @Ylianst has brought a brilliant ideas to the feature and it's clear that he add a very nice touch and important options there, like ability to drag and drop whatever file to run. also the option to choose between run as agent or pc or user.

at the end I wish he stick to one page style like how the plugin is, without complicated dialogs and menus while still improving the functionality and options and add his touch like the above features he already achieved. That would be a system on it's own, very strong and 100% preferred and wanted.

I wish I didn't offend anyone with the above, if I did then please accept my apology in advance. I tried my best to explain how wish it to be for the best for all. others maybe disagree with me, but at least I expressed my thoughts of it.

Thanks all and sorry for the long message.

**Comment by exander77** [comment]
> > Can We expect some improvement to get result from each node?
> 
> yes @Ylianst said it was in the works but for the moment its only runs commands he is away at the moment, but the plan he said was to merge the scripttask plugin into MC

Do you have a timeline? Also, it would be great that if PC is not turned on, the command waits in a queue, and it will run as soon as PC is on.

**Comment by exander77** [comment]
@Ylianst Enjoy the holiday!
